src/ManifestStrategy/WireNavigationManifestStrategy.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\ManifestStrategy;
+
+use SimpleXMLElement;
+use SprykerSdk\Integrator\Dependency\Console\InputOutputInterface;
+
+class WireNavigationManifestStrategy extends AbstractManifestStrategy
+{
+    /**
+     * @var string
+     */
+    protected const TARGET_NAVIGATION_FILE = 'config/Zed/navigation.xml';
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return 'wire-navigation';
+    }
+
+    /**
+     * @param array<mixed> $manifest
+     * @param string $moduleName
+     * @param \SprykerSdk\Integrator\Dependency\Console\InputOutputInterface $inputOutput
+     * @param bool $isDry
+     *
+     * @return bool
+     */
+    public function apply(array $manifest, string $moduleName, InputOutputInterface $inputOutput, bool $isDry): bool
+    {
+        if (!file_exists(static::TARGET_NAVIGATION_FILE)) {
+            $inputOutput->writeln(sprintf(
+                'The project doesn\'n have the navigation source file: %s.',
+                static::TARGET_NAVIGATION_FILE,
+            ), InputOutputInterface::DEBUG);
+
+            return false;
+        }
+
+        $mainNavigationXmlElement = simplexml_load_file(static::TARGET_NAVIGATION_FILE);
+
+        foreach ($manifest['navigation'] as $navigationKey => $navigationData) {
+            //TODO::Should it (the injection of navigation into the nested exist navigation item) be implemented now?
+            // Check cthe comment inside the getTargetNavigationXmlElement method.
+//            if (strpos($navigationKey, '.') === false) {
+//                $targetXmlElement = $mainNavigationXmlElement;
+//            } else {
+//                $targetXmlElement = $this->getTargetNavigationXmlElement($mainNavigationXmlElement, $navigationKey);
+//                $navigationKey = explode('.', $navigationKey)[0];
+//            }
+            $targetXmlElement = $mainNavigationXmlElement;
+
+            //TODO::check if navigation is present and skip if it is (check it in nested group of navigation)
```

Comment:
You could use `$xml->xpath(...)` to check if the element exists before you iterate over the XML elements and return early if it exists.